osier/models/capacity_expansion.py

```diff
@@ -0,0 +1,176 @@
+# Standard imports
+import numpy as np
+import pandas as pd
+from copy import deepcopy
+import dill
+
+# Osier imports
+from osier import DispatchModel
+
+# Pymoo imports
+from pymoo.core.problem import Problem
+from pymoo.core.problem import ElementwiseProblem
+from pymoo.algorithms.moo.nsga2 import NSGA2
+from pymoo.termination import get_termination
+from pymoo.optimize import minimize
+from pymoo.termination.ftol import MultiObjectiveSpaceTermination
+from pymoo.visualization.scatter import Scatter
+from pymoo.operators.sampling.rnd import FloatRandomSampling
+from pymoo.operators.crossover.sbx import SBX
+from pymoo.operators.mutation.pm import PolynomialMutation
+from pymoo.termination.robust import RobustTermination
+from pymoo.core.parameters import set_params, hierarchical
+
+
+LARGE_NUMBER = 1e40
+
+class CapacityExpansion(ElementwiseProblem):
+    """
+    The :class:`CapacityExpansion` class inherits from the 
+    :class:`pymoo.core.problem.ElementwiseProblem` class. This problem
+    determines the technology mix that _minimizes_ the provided
+    objectives.
```

Comment:
Maybe we can massage this a little bit. It's not obvious to me what the `CapacityExpansion` class represents just from reading this docstring.

Reply:
Hmmm this class is just a `problem` that specifies how the decision space gets mapped onto objective space. This is not itself a genetic algorithm -- it's the map. This requires a bit of knowledge about how `pymoo` works but at this level I think that's a safe assumption (I'm slowly adding layers of abstraction and this layer is not the most abstract layer). I'm definitely open to ideas.

Reply:
The way I would do it is describe in general terms what `CapacityExpansion` is in terms of how it is used. Based on the name, perhaps something like the following would work: "This class represents a problem where the objective is to expand the capacity of technologys to a user-defined level". It's possible I'm misunderstanding this.

Reply:
The phrase "objective" has a pretty strict definition here and I don't want to confuse people. The "objective" for a user is to identify a solution set (obv) but the "objective(s)" of the problem are user-defined ("provided objectives"). I think the doc string is clear enough.

Reply:
Okay. I think the confusion is just from my relative lack of familiarity with optimization terminology.